File: app/schemas/scoring.py
from datetime import datetime

from pydantic import BaseModel


class ScoringRuleCreate(BaseModel):
    field: str
    operator: str
    value: str
    points: int
    is_active: bool = True


class ScoringRuleUpdate(BaseModel):
    field: str | None = None
    operator: str | None = None
    value: str | None = None
    points: int | None = None
    is_active: bool | None = None


class ScoringRuleResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: int
    client_id: int
    field: str
    operator: str
    value: str
    points: int
    is_active: bool
    created_at: datetime
    updated_at: datetime


class ScoringRuleListResponse(BaseModel):
    items: list[ScoringRuleResponse]
    total: int


class ScoringTemplateRule(BaseModel):
    field: str
    operator: str
    value: str
    points: int


class ScoringTemplateResponse(BaseModel):
    name: str
    description: str
    rules: list[ScoringTemplateRule]

File: frontend/src/types/scoring.ts
export interface ScoringRule {
  id: number;
  client_id: number;
  field: string;
  operator: string;
  value: string;
  points: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ScoringRuleListResponse {
  items: ScoringRule[];
  total: number;
}

export interface ScoringTemplateRule {
  field: string;
  operator: string;
  value: string;
  points: number;
}

export interface ScoringTemplate {
  name: string;
  description: string;
  rules: ScoringTemplateRule[];
}

export const VALID_OPERATORS = [
  "equals",
  "not_equals",
  "contains",
  "not_contains",
  "greater_than",
  "less_than",
  "not_empty",
] as const;

export const OPERATOR_LABELS: Record<string, string> = {
  equals: "Equals",
  not_equals: "Not Equals",
  contains: "Contains",
  not_contains: "Not Contains",
  greater_than: "Greater Than",
  less_than: "Less Than",
  not_empty: "Not Empty",
};
